features/frameworks/utest/source/utest_harness.cpp

```
@@ -165,9 +165,11 @@ bool Harness::run(const Specification& specification)
 void Harness::raise_failure(const failure_reason_t reason)
 {
     UTEST_LOG_FUNCTION();
-    // ignore a failure, if the Harness has not been initialized.
-    // this allows using unity assertion macros without setting up utest.
-    if (test_cases == NULL) return;
```

This old behavior (silently do nothing if not inside a test case) causes some issues with tests that do assertions but don't interact with utest at all (they do exist!).  With the old behavior, such test would print a "failure" message but not actually fail!  So the failure would get ignored for all time.